cpu/rp2350/clock.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Tom Hert <git@annsann.eu>
+ * SPDX-FileCopyrightText: 2025 HAW Hamburg
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+#include "periph_cpu.h"
+
+void clock_reset(void) {
+  /* Reset the clock system */
+  reset_component(RESET_PLL_SYS, RESET_PLL_SYS);
+}
+
+/**
+ * @brief Configures the XOSC and then sets CLK_SYS, PLL_SYS and CLK_PERI to it
+ * @warning Make sure to call clock_reset() before this function to reset the
+ * clock system
+ * @note RP2350 Docs Chapter 8, mostly 8.2 for more details
+ */
+void cpu_clock_init(void) {
+  /* Enable the XOSC */
+  xosc_start();
+
+  /**
+   * Setup the PLL using the XOSC as the reference clock.
+   */
```

Comment:
```suggestion
  /* Setup the PLL using the XOSC as the reference clock. */
```

This does not really have to be a Doxygen block comment, as Doxygen does not evaluate the internals of functions.